compiler/plugins/target/AMD-AIE/aie/AIEOps.td

```diff
@@ -551,6 +550,20 @@ def AIE_ObjectFifoCreateOp: AIE_Op<"objectfifo", [Symbol]> {
     mlir::StringAttr name() {
       return ::xilinx::AIE::name(*this);
     }
+
+    int size(int index = 0) {
+      auto elemNumber = getElemNumber();
+      auto intAttr = [&]() -> mlir::IntegerAttr {
+        if (auto arrAttr = llvm::dyn_cast<mlir::ArrayAttr>(elemNumber))
+          return llvm::dyn_cast<mlir::IntegerAttr>(arrAttr[index]);
+        else
+          return llvm::dyn_cast<mlir::IntegerAttr>(elemNumber);
+      }();
+      assert(intAttr && "expected integer attribute");
+      return intAttr.getInt();
+    }
```

Comment:
I deleted this thing and replaced it with `objFifoSize` because 

1) putting cpp into tablegen effectively hides it (eg completely confuses `clangd`); 
2) member methods also hide the side effects of the function;
3) single use functions like this should be kept as close to their use as possible.

This is roughly inline with google's and various other style guides that recommend private (anon namespace) by default.